### PR TITLE
Dremio task now uses arrow.flight.auth2 logic

### PR DESCRIPTION
**Pull request by strojank**
Dremio task `DremioFetch` now `uses arrow.flight.auth2` logic instead of `legacy.arrow.flight.auth`

## Summary

Dremio is deprecating legacy Arrow Flight authentication. This PR updates the logic to be in accordance with default Dremio settings.

## Changes

`DremioFetch` task now uses different middleware from `pyarrow`. Legacy mode is not supported. This introduces breaking change for existing users of `DremioFetch`. 

## Importance

It updates Dremio tasks to work with latest Dremio versions.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/tasks/dremio/dremio.py [hunk 1]
```diff
@@ -84,14 +127,80 @@ def run(self, user: str, password: str, host: str, port: int, query: str) -> Dic
...
-        client = flight.FlightClient(f"grpc+tcp://{host}:{port}")
-        client.authenticate(HttpDremioClientAuthHandler(user, password))
-
-        info = client.get_flight_info(
-            flight.FlightDescriptor.for_command(query + "--arrow flight")
+        # Authenticate with the server endpoint.
+        bearer_token = client.authenticate_basic_token(
+            username, password, initial_options
+        )
+        print("[INFO] Authentication was successful. Token is valid for 30 hours.")
```
Comment (madkinsz) [inline src/prefect/tasks/dremio/dremio.py:173]: Can you use `self.logger` instead so these will show up in the UI?

File: src/prefect/tasks/dremio/dremio.py [hunk 2]
```diff
@@ -84,14 +127,80 @@ def run(self, user: str, password: str, host: str, port: int, query: str) -> Dic
...
+
+        flight_desc = flight.FlightDescriptor.for_command(query)
+        print("[INFO] Query: ", query)
+
+        # In addition to the bearer token, a query context can also
+        # be provided as an entry of FlightCallOptions.
+        # options = flight.FlightCallOptions(headers=[
+        #     bearer_token,
+        #     (b'schema', b'test.schema')
+        # ])
```
Comment (madkinsz) [inline src/prefect/tasks/dremio/dremio.py:183]: What's your plan for this?

File: src/prefect/tasks/dremio/dremio.py (hunk 1 above)
Comment (madkinsz) [reply src/prefect/tasks/dremio/dremio.py:173]: A bunch of this seems DEBUG level

File: tests/tasks/dremio/test_dremio.py [hunk 3]
```diff
@@ -1,20 +1,47 @@
...
 from pyarrow.flight import FlightUnavailableError
 
 from prefect.tasks.dremio.dremio import DremioFetch
 
 
 class TestDremioFetch:
     def test_construction(self) -> None:
-        task = DremioFetch(user="test", password="test", host="test")
-        assert (task.user == "test") and (task.host == "test")
+        task = DremioFetch(username="test", password="test", hostname="test")
```
Comment (madkinsz) [inline tests/tasks/dremio/test_dremio.py:10]: I prefer to pass different values for each of these so your test ensures they are set correctly

**Review by madkinsz — COMMENTED** [review]
Thanks for tackling this. A few comments.

File: src/prefect/tasks/dremio/dremio.py (hunk 2 above)
Comment (strojank) [reply src/prefect/tasks/dremio/dremio.py:183]: I would leave this out for now.

File: src/prefect/tasks/dremio/dremio.py (hunk 1 above)
Comment (strojank) [reply src/prefect/tasks/dremio/dremio.py:173]: Thanks for your feedback @madkinsz. I used `self.logger` where appropriate and removed redundant print statements.

File: tests/tasks/dremio/test_dremio.py (hunk 3 above)
Comment (strojank) [reply tests/tasks/dremio/test_dremio.py:10]: Good call. Done.

**Review by madkinsz — APPROVED** [review]

**Comment by madkinsz** [comment]
Looking forward to getting this released!

**Pull request merged**